### ISSUE TITLE
Refactor options validation

### ISSUE THREAD

**Issue by fgreinacher** [semantic-release/gitlab#368]
<a href='https://hearts.dev/projects/3/users/fgreinacher'><img width='50' alt='' align='right' src='https://hearts.dev/projects/3/users/fgreinacher/tally.svg'></a>

@travi @JonasSchubert I'd suggest to tackle this in two steps:

1.  Merge this fix to unblock folks asap.
2. Create a follow-up issue to refactor the option validation. IMO it's super brittle and we should clean it up properly so that we cannot run into these kinds of issue in the future. We should somehow encapsulate options, their validation and error messages better.

__Originally posted by @fgreinacher in https://github.com/semantic-release/gitlab/pull/364#issuecomment-1098849915__

**Comment by JonasSchubert** [comment]
How shall we continue here?
To keep it similar to semantic-release/github I simply would add the missing validations and write a Unit Test checking whether all validations are covered (Including new validations).
Or did you consider a bigger refactoring @fgreinacher ?

**Comment by fgreinacher** [comment]
Probably not worth it right now. We haven't seen new issues for quite some time.
